File: app/src/main/java/com/reitler/got/model/statistics/PlayerStatistic.java
package com.reitler.got.model.statistics;

import com.reitler.got.model.data.entity.PlayerEntity;

import java.util.Comparator;
import java.util.LinkedList;
import java.util.List;

public class PlayerStatistic {

    private PlayerEntity player;
    private List<Integer> scores = new LinkedList<>();

    public PlayerStatistic(PlayerEntity entity) {
        this.player = entity;
    }

    public void addScore(int remainingScore){
        this.scores.add(remainingScore);
    }

    public int getMatchCount(){
        return this.scores.size();
    }

    public int getWinCount(){
        return (int) this.scores.stream().filter(i -> i == 0).count();
    }

    public int highestRemaining(){
        return this.scores.stream().max(Comparator.naturalOrder()).get();
    }
}

File: app/src/main/java/com/reitler/got/model/statistics/PlayerStatistic.kt
package com.reitler.got.model.statistics

import com.reitler.got.model.data.entity.PlayerEntity
import java.util.*
import java.util.stream.Collectors

class PlayerStatistic(val player: PlayerEntity) {
    private val scores: MutableList<Int> = LinkedList()
    fun addScore(remainingScore: Int) {
        scores.add(remainingScore)
    }

    val matchCount: Int
        get() = scores.size
    val winCount: Int
        get() = scores.count { it == 0 }
    val highestRemaining: Int
        get() = if (matchCount == 0) 0 else scores.maxOf { it }
    val averageRemaining: Double
        get() = if (matchCount == 0) 0.0 else scores.sum().toDouble() / (matchCount)
    val totalRemaining: Int
        get() = scores.sum()
}